woodwork/type_sys/type_system.py

```diff
@@ -252,16 +254,25 @@ def infer_logical_type(self, series):
         Args:
             series (pandas.Series): The series for which to infer the LogicalType.
         """
+        if isinstance(series, pd.Series):
+            # Special case for series with no valid values
+            if series.count() == 0:
+                return Unknown()
 
-        # Bring Dask or Koalas data into memory for inference
-        if _is_dask_series(series):
-            series = series.get_partition(0).compute()
-        if _is_koalas_series(series):
-            series = series.head(100000).to_pandas()
+            series = series.head(100000)
```

Comment:
Should we store 100000 in a variable so if we decide to change it in the future, we only have to remember to do it in one place?

Reply:
Sure, y not

Reply:
By the way, I know I did the same thing with the constant in `inference_functions`.  In this case, it seemed like the values were all within a few lines of each other so it didn't matter as much.  But I think I was also sort of just being lazy :).